tools/src/main/scala/scala/scalanative/linker/Infos.scala

```diff
@@ -159,7 +159,9 @@ final class Class(val attrs: Attrs,
       }
 
     add(this)
-    subclasses.foreach(add)
+    if (sig.isVirtual || out.isEmpty) {
```

Comment:
It's still unclear for me why the `|| out.isEmpty` condition was needed. When it's omitted program resulted in segfaults.

Reply:
That doesn't look good. It doesn't really make sense. Do you have any idea what kind of code pattern cause the segfault without that condition?

Reply:
Here's one example which gives errors in Check
```
[warn]   0000  didopt def @"M17java.lang.System$D14loadPropertiesL20java.util.PropertiesEPT17java.lang.System$" : (@"T17java.lang.System$") => nothing {
[warn]   0001  %20000(%1 : @"T17java.lang.System$"):
[warn]   0002    %50001 = module @"T33scala.collection.mutable.HashMap$"
[warn]   0003    %50002 = call[(@"T33scala.collection.mutable.HashMap$") => @"T32scala.collection.mutable.HashMap"] @"M33scala.collection.mutable.HashMap$D5emptyL32scala.collection.mutable.HashMapEO" : ptr(%50001 : !?@"T33scala.collection.mutable.HashMap$")
[warn]   0004    %50003 = classalloc @"T20java.util.Properties"
[warn]   0005    %50004 = method %50003 : !?@"T20java.util.Properties", "RL32scala.collection.mutable.HashMapE"
[warn]   0006    unreachable

...
[warn]   in inst #5 : 
[warn]     can't call RL32scala.collection.mutable.HashMapE on @"T20java.util.Properties"
```

`Properties` does not have such constructor, but it extends `Hashtable` which does. It happens after optimization stage so probably it was inlined for some reason. 

It ends up with exit code 134 instead of seg fault